test/integration/smoke/test_router_dns.py

```diff
@@ -0,0 +1,286 @@
+
+        cls.services["natrule1"] = {
+            "privateport": 22,
+            "publicport": 22,
+            "protocol": "TCP"
+        }
+
+        cls.services["configurableData"] = {
+            "host": {
+                "password": "password",
+                "username": "root",
+                "port": 22
+            },
+            "input": "INPUT",
+            "forward": "FORWARD"
+        }
+
+        cls._cleanup = [
+            cls.vm,
+            cls.network,
+            cls.network_offering,
+            cls.service_offering,
+            cls.account
+        ]
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.api_client, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+
+    @attr(tags=["advanced", "advancedns", "ssh"], required_hardware="true")
+    def test_router_dns_externalipquery(self):
+        """Checks that non-guest network IPs cannot access VR DNS"""
+
+        self.logger.debug("Starting test_router_dns_externalips...")
+        routers = list_routers(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid
+        )
+
+        self.assertEqual(
+            isinstance(routers, list),
+            True,
+            "Check for list routers response return valid data"
+        )
+
+        self.assertNotEqual(
+            len(routers),
+            0,
+            "Check list router response"
+        )
+
+        router = routers[0]
+
+        self.assertEqual(
+            router.state,
+            'Running',
+            "Check list router response for router state"
+        )
+
+        public_ips = list_publicIP(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid,
+            zoneid=self.zone.id
+        )
+
+        self.assertEqual(
+            isinstance(public_ips, list),
+            True,
+            "Check for list public IPs response return valid data"
+        )
+
+        public_ip = public_ips[0]
+        self.logger.debug("Querying VR DNS IP: " + public_ip.ipaddress)
+        resolver = dns.resolver.Resolver()
+        resolver.namerservers = [public_ip.ipaddress]
+        try:
+            resolver.query('google.com', 'A')
+            self.fail("Non-guest network IPs are able to access VR DNS, failing.")
+        except:
+            self.logger.debug("VR DNS query failed from non-guest network IP as expected")
+
+
+    @attr(tags=["advanced", "advancedns", "ssh"], required_hardware="true")
+    def test_router_dns_guestipquery(self):
```

Comment:
The setup of this test method and ``test_router_dns_externalipquery`` seem very similar.  Would it make sense to factor it out to the ``setup`` method or some other internal utility method?